cmd/check-cluster-profiles-config/main.go

```diff
@@ -0,0 +1,107 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"os"
+
+	"github.com/sirupsen/logrus"
+
+	coreapi "k8s.io/api/core/v1"
+	ctrlruntimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/yaml"
+
+	"github.com/openshift/ci-tools/pkg/api"
+	"github.com/openshift/ci-tools/pkg/util"
+)
+
+type options struct {
+	configPath string
+}
+
+type profileValidator struct {
+	profiles   api.ClusterProfilesMap
+	kubeClient ctrlruntimeclient.Client
+}
+
+func gatherOptions() options {
+	o := options{}
+	fs := flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+
+	fs.StringVar(&o.configPath, "config-path", "", "Path to the cluster profile config file")
+
+	if err := fs.Parse(os.Args[1:]); err != nil { // todo: should this be args[1:] ?
+		logrus.WithError(err).Fatal("could not parse arguments")
+	}
+
+	return o
+}
+
+func newValidator() *profileValidator {
+	inClusterConfig, err := util.LoadClusterConfig()
+	if err != nil {
+		logrus.WithError(err).Fatal("failed to load in-cluster config")
+	}
+
+	client, err := ctrlruntimeclient.New(inClusterConfig, ctrlruntimeclient.Options{})
+	if err != nil {
+		logrus.WithError(err).Fatal("failed to create client")
+	}
+
+	return &profileValidator{
+		profiles:   make(api.ClusterProfilesMap),
+		kubeClient: client,
+	}
+}
+
+func main() {
+	logger := logrus.WithField("component", "check-cluster-profiles-config")
+	o := gatherOptions()
+
+	validator := newValidator()
+	if err := validator.loadConfig(o.configPath); err != nil {
+		logger.WithError(err).Fatal("failed to load profiles from config")
+	}
+
+	if err := validator.checkCiSecrets(); err != nil {
+		logger.WithError(err).Fatal("failed to check secrets for cluster profiles")
+	}
+
+	// TODO after they are implemented into the config file, add checks for cluster type and leases
```

Comment:
This is a personal TODO with no details. Let's remove it. I am sure that you will follow-up on the missing features either way.